include/ginkgo/core/base/executor.hpp

```diff
@@ -511,6 +511,31 @@ class Executor : public log::EnableLogging<Executor> {
         this->run(op);
     }
 
+    /**
+     * Runs one of the passed in functors, depending on the Executor type
+     * with a custom operation name.
+     *
+     * @tparam ClosureOmp  type of op_omp
+     * @tparam ClosureCuda  type of op_cuda
+     * @tparam ClosureHip  type of op_hip
+     *
+     * @param name  the name to be used for the underlying operation
+     * @param op_omp  functor to run in case of a OmpExecutor or
+     *                ReferenceExecutor
+     * @param op_cuda  functor to run in case of a CudaExecutor
+     * @param op_hip  functor to run in case of a HipExecutor
```

Comment:
missing dpcpp

```suggestion
     * @param op_hip  functor to run in case of a HipExecutor
     * @param op_dpcpp  functor to run in case of a DpcppExecutor
```